examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -292,21 +315,40 @@ def score_estimator(estimator, df_test):
 
 ##############################################################################
 # The experimental data presents a long tail distribution for ``y``. In all
-# models we predict a mean expected value, so we will have necessarily fewer
-# extreme values. Additionally, the normal distribution used in ``Ridge`` and
-# ``RandomForestRegressor`` has a constant variance, while for the Poisson
-# distribution used in ``PoissonRegressor``, the variance is proportional to
-# the mean predicted value.
+# models, we predict the expected frequency of a random variable, so we will
+# have necessarily fewer extreme values than for the observed realizations of
+# that random variable. Additionally, the normal distribution used in ``Ridge``
+# has a constant variance, while for the Poisson distribution used in
+# ``PoissonRegressor`` and ``HistGradientBoostingRegressor``, the variance is
+# proportional to the predicted expected value.
+#
+# Thus, among the considered estimators, ``PoissonRegressor`` and
+# ````HistGradientBoostingRegressor```` are a-priori better suited for modeling
+# the long tail distribution of the non-negative data as compared to the
+# ``Ridge`` model which makes a wrong assumption on the distribution of the
+# target variable.
 #
-# Thus, among the considered estimators, ``PoissonRegressor`` is better suited
-# for modeling the long tail distribution of the data as compared to the
-# ``Ridge`` and ``RandomForestRegressor`` estimators.
+# The ``HistGradientBoostingRegressor`` estimator has the most flexibility and
+# is able to predict higher expected values.
+#
+# Note that we could have used the least squares loss for the
+# ``HistGradientBoostingRegressor`` model. This would wrongly assume a normal
+# distribution the response variable as for the `Ridge` model and possibly also
+# lead to slightly negative predictions. However the gradient boosted trees
+# would still perform relatively well and in particular better than
+# ``PoissonRegressor`` thanks to the flexibility of the trees combined with
+# the large number of training samples.
+#
+# Evaluation of the calibration of predictions
```

Comment:
Do the curves that are plotted below have a specific name?

Surely they are different from calibration curves as we have them in the [UG](https://scikit-learn.org/stable/modules/calibration.html#calibration) ?

Reply:
The calibration curves currntly in the user guide are only defined for classification problems. They are similar to those int this example (plot the mean observed values per prediction bin) when using the quantile binning strategy.

It would make sense to extend the `calibration_curve` tool to support regression but this should be done in a dedicated PR.